src/test/java/io/asyncer/r2dbc/mysql/MySqlConnectionFactoryProviderTest.java

```diff
@@ -450,6 +453,44 @@ void validPasswordSupplier() {
         assertThat(ConnectionFactories.get(options)).isExactlyInstanceOf(MySqlConnectionFactory.class);
     }
 
+    @Test
+    void allConfigurationOptions() {
+        List<String> exceptConfigs =  Arrays.asList(
+            "extendWith",
+            "username",
+            "zeroDateOption");
+        List<String> exceptOptions = Arrays.asList(
+            "driver",
+            "ssl",
+            "protocol",
+            "zeroDate",
+            "lockWaitTimeout",
+            "statementTimeout");
+        Set<String> allOptions = Stream.concat(
+                Arrays.stream(ConnectionFactoryOptions.class.getFields()),
+                Arrays.stream(MySqlConnectionFactoryProvider.class.getFields())
+            )
+            .filter(field -> Modifier.isStatic(field.getModifiers()) && field.getType() == Option.class)
+            .map(field -> {
+                try {
+                    return ((Option<?>) field.get(null)).name();
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                }
+            })
+            .filter(name -> !exceptOptions.contains(name))
+            .collect(Collectors.toSet());
+        Set<String> allBuilderOptions = Arrays.stream(MySqlConnectionConfiguration.Builder.class.getMethods())
+            .filter(method -> method.getParameterCount() >= 1 &&
+                method.getReturnType() == MySqlConnectionConfiguration.Builder.class &&
+                !exceptConfigs.contains(method.getName()))
+            .map(Method::getName)
+            .collect(Collectors.toSet());
+
+        assertThat(allOptions).containsAll(allBuilderOptions);
+        assertThat(allBuilderOptions).containsAll(allOptions);
```

Comment:
Could we consider replacing it like below?
`assertThat(allOptions).containsExactlyInAnyOrderElementsOf(allBuilderOptions);`